src/client/api/restful/send_tagged_data.c

```diff
@@ -0,0 +1,182 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdio.h>
+
+#include "client/api/json_parser/json_utils.h"
+#include "client/api/restful/get_tips.h"
+#include "client/api/restful/send_tagged_data.h"
+#include "core/utils/macros.h"
+#include "utarray.h"
+
+int send_tagged_data_message(iota_client_conf_t const* conf, byte_t tag[], uint8_t tag_len, byte_t data[],
+                             uint32_t data_len, res_send_message_t* res) {
+  if (conf == NULL || tag == NULL || res == NULL) {
+    // invalid parameters
+    printf("[%s:%d] invalid parameters\n", __func__, __LINE__);
+    return -1;
+  }
+
+  if (data_len > 0 && data == NULL) {
+    printf("[%s:%d] data cannot be null if data_len is greater than 0\n", __func__, __LINE__);
+    return -1;
+  }
+
+  // Max tag length allowed is 64
+  if (tag_len > 64) {
```

Comment:
Use `#define` instead of a hardcoded value.

Reply:
Done.